### DISCUSSION TITLE
Chapter 03. 데이터베이스 시스템

### DISCUSSION THREAD

**Discussion by new-pow** [CodeSquad-2023-BE-Study/DB-Study#4] [완료]
책을 읽으면서 궁금한 내용을 남겨주세요.

**Comment by jinny-l** [comment]
스키마를 3단계로 나눈 개념이 대략적으로 느낌만 오고 잘 와닿지 않네요 🥲

**Reply by new-pow** [reply]
저도 객체지향의 캡슐화와 비슷한 느낌적인 느낌으로 이해했습니다.
개념단계에 대한 이해가 더 필요할 것 같아보입니다.

**Reply by new-pow** [reply]
p. 69 에 대한 내용임

**Reply by crtEvent** [reply]
데이터 웨어하우스 <- 외부 스키마 많이 쓰임. 운영할 때 많이 쓰임??

**Comment by jinny-l** [comment]
`P78`
- SQL은 선언적 언어로 알고 있는데, 절차적 조작어도 있다는 내용이 보여 검색해 보았습니다.
- 프로시저나 트리거, 사용자 정의 함수가 절차적 조작어가 속하는 것으로 이해했습니다.

이전에 호눅스가 프로시저에 대해 강의할 때 절차적 vs 선언적에 대해 설명해주신 것이 기억이 나네요

**Comment by jinny-l** [comment]
데이터베이스 시스템은 제일 넓은 범위의 개념으로 "사용자 → 데이터베이스 관리 시스템 → 데이터베이스 → 컴퓨터"까지 모두 아우르는 개념으로 이해했습니다.

그리고 MySQL, Oracle 이런 소프트웨어들은 "데이터베이스 관리 시스템"으로 이해했는데 맞게 이해했는지..
DBMS는 정확히 어떤 것인지 헷갈리네요,

유독 3장은 모호한 개념이 많게 느껴지네요

**Reply by crtEvent** [reply]
저도 이렇게 이해했습니다. 

저는 MySQL, Oracle 같은 소프트웨어가 DB라고 생각했는데 이것들이 DBMS이고

DB는 그냥 데이터의 집합, 데이터의 모음 이라는 걸 처음 알았습니다.

**Reply by new-pow** [reply]
저도 똑같이 이해했습니다. 추상적인 개념 나열로 끝나서 1~2장 읽었을 때와 비슷한 느낌이긴 하네요 ㅎㅎ
